### ISSUE TITLE
[feature request] respect kubernetes health checks

### ISSUE THREAD

**Issue by afoninsky** [keel-hq/keel#336]
Is it possible to trigger notifications using custom webhook after deployment success/fail based on deployment health check? For now it triggers on resource apply and does not wait for successful health check. So we can embed keel into CI/CD pipeline more natively.

Also it will be useful to implement webhook notification on new approval request (it works only for slack now).

**Comment by rusenask** [comment]
Yeah, for sure! Would love to add that :) There's already some code from the bot:
```
NAMESPACE           NAME                    READY               IMAGES
default             curator                 1/1                 gcr.io/webhookrelay/curator:latest, gcr.io/cloudsql-docker/gce-proxy:1.09
default             grafana                 1/1                 grafana/grafana:4.6.3
```
That gets 'ready' pods, we could have something similar, wait for a bit and check what's the situation with pods. Since all of that info is already in the cache with K8s provider, checking it becomes very cheap.

**Comment by afoninsky** [comment]
Cool, thank you :) And what about my second proposal (custom approval request)? Is it possible? As I see there is a webhook for slack exists but custom one is not present.

My use case is: to receive requests for updating k8s services (databases, etc) using any devops tools. Implementation example: chatbot via https://stackstorm.com/ and https://github.com/afoninsky/stackstorm-keel

**Comment by rusenask** [comment]
Hi, started looking into the approval webhooks. I think it would be good to have them, but not sure whether it fits into the existing notification system as it's more suitable for generic system info. Would a separate environment variable for approval webhooks be a better solution?

**Comment by afoninsky** [comment]
If it suits you - sound great :)
